dialogue-core/src/main/java/com/palantir/dialogue/core/Config.java

```diff
@@ -25,11 +25,22 @@
 import java.util.OptionalInt;
 import java.util.Random;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
 import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Private class to centralize validation of params necessary to construct a dialogue channel. */
 @Value.Immutable
 interface Config {
+    /**
+     * This prefix may reconfigure several aspects of the client to work better in a world where requests are routed
+     * through a service mesh like istio/envoy.
+     */
+    String MESH_PREFIX = "mesh-";
+
+    Logger log = LoggerFactory.getLogger(Config.class);
```

Comment:
Not a fan of public loggers. Can we do this differently?

Reply:
Eh, I suppose the interface is internal and package private. Slight preference to use an immutabels abstract class to encapsulate this sort of thing.

Reply:
The logger should go away if we fail hard when we have a mixture of mesh and non-mesh uris

Reply:
Think this is resolved now